CAVEATS.md

```diff
@@ -0,0 +1,232 @@
+<!--
+    Copyright 2014, General Dynamics C4 Systems
+
+    SPDX-License-Identifier: GPL-2.0-only
+-->
+
+# Known caveats in the seL4 API and implementation
+
+## Implementation Correctness
+
+The following seL4 architectures have platforms with a C-level functional
+correctness proof. Proof support for further platforms within these
+architectures is on the roadmap and expected in 2025.
+
+- AArch32: Armv7-a with and without hypervisor extensions, no SMMU, with
+  fast path
+  - Platforms (non-hyp): `sabre` (no FPU), `imx8mm-evk` (with FPU)
+  - Platforms (hyp, no FPU): `tk1`, `exynos5`
+- AArch64: Armv8-a with hypervisor extensions only, no SMMU, with fast path
+  - Platforms: `tx2`
+- RISC-V: 64-bit only, no fast path
+  - Platforms: `hifive`
+- x64: without VT-x and VT-d, no fast path
+  - Platforms: `pc99`
+
+This proof covers the functional behaviour of the C code of the kernel. It does
+not cover machine code, compiler, linker, boot code, cache and TLB management.
+Compiler and linker can be removed from this list by additionally running the
+binary verification tool chain for seL4 for AArch32 or RISC-V. The proof shows
+that the seL4 C code implements the formal [abstract API specification][ASpec]
+of seL4 and is free from standard C implementation defects such as buffer
+overruns or NULL pointer dereferences.
+
+For AArch32 without hypervisor extensions and without FPU, and for RISC-V, there
+are additional proofs that this specification satisfies the following high-level
+security properties:
+
+- integrity (no write without authority),
+- confidentiality (no read without authority), and
+- intransitive non-interference (isolation, modulo timing channels, between
+  adequately configured user-level components).
+
+The security property proofs depend on additional assumptions on the correct
+configuration of the system. See the [l4v] repository on GitHub for more
+details.
+
+Similar proofs for AArch64 with hypervisor extensions are in progress.
+
+For AArch32, there additionally exist proofs for correct user-level system
+initialisation. See the [l4v] repository for details.
+
+Note that seL4 currently performs lazy FPU and VCPU switching, which can
+introduce information flow timing channels. An API-change proposal ([RFC]) to
+improve this behaviour is currently in progress.
+
+## Verified Configurations
+
+For the precise configuration of the verified platforms above, see the
+corresponding files in the seL4 `configs/` directory.
+
+The proofs are generally sensitive to changes in configuration parameters, and
+will break if these are changed. For some parameters, the proofs are explicitly
+set up to be robust, such as the number of domains `NUM_DOMAINS`, and the domain
+schedule. More such parameters are on the roadmap to be added and documented
+here.
```

Comment:
is your maxIRQ stuff in, or too early?

Reply:
also not merged yet, but looking forward to adding it to the next release notes :-)